### PR TITLE
make custom worker, eventListener and throwableHandler work again

### PR DESCRIPTION
**Pull request by peh**
with the change to .yml based configration we have to expect the custom clazzes to be defined as string.
so i implemented a way to do this as strings OR classes.

i also found a severe bug in GrailsWorkerImpl which was calling jedis methods on the jedisPool instead of a ressource from within the pool.

### CODE REVIEW

File: src/main/groovy/grails/plugins/jesque/GrailsWorkerImpl.groovy [hunk 1]
```diff
@@ -82,63 +82,16 @@ class GrailsWorkerImpl extends WorkerImpl {
...
             log.info "Running perform on instance ${job.className}"
             final Object result
             this.listenerDelegate.fireEvent(JOB_EXECUTE, this, curQueue, job, instance, null, null)
             result = instance.perform(*args)
             success(job, instance, result, curQueue)
         } finally {
-            this.jedis.del(key(WORKER, this.name))
+            this.jedisClient.del(key(WORKER, this.name))
         }
     }
```
Comment (peh) [inline src/main/groovy/grails/plugins/jesque/GrailsWorkerImpl.groovy]: as now everything is pool based, we do not have to care about getting a fresh connection but can rely on the pool to handle all that for us

File: src/main/groovy/grails/plugins/jesque/WorkerImpl.java [hunk 2]
```diff
@@ -772,6 +772,10 @@ protected String lpoplpush(final String from, final String to) {
         return result;
     }
```
Comment (peh) [inline src/main/groovy/grails/plugins/jesque/WorkerImpl.java]: helper method to grab a JedisClient from the pool

File: grails-app/services/grails/plugins/jesque/JesqueService.groovy [hunk 3]
```diff
@@ -130,29 +130,58 @@ class JesqueService implements DisposableBean {
         Class workerClass = GrailsWorkerImpl
         def customWorkerClass = grailsApplication.config.grails.jesque.custom.worker.clazz
         if (customWorkerClass) {
-            if (customWorkerClass in GrailsWorkerImpl) {
+            if (customWorkerClass instanceof String) {
```
Comment (peh) [inline grails-app/services/grails/plugins/jesque/JesqueService.groovy]: the actual class getting could be refactored and dry'ed a bit

**Comment by ctoestreich** [comment]
@peh Thanks.  Looks like a missed that one pool change.  I wasn't a huge fan of changing to a pool locally as it meant I had to pull in and change the core classes from Jesque.  I have a pseudo PR (sent him my changed code) open with Greg Haines who owns the core jesque library.  Using the pool is way better IMHO and to support Sentinel it was the only way given how the connection used to be created.

**Comment by peh** [comment]
yeah back in the grails-2 days i tried to do the same and we have been stuck with a custom flat file dependency to our own jesque & grails-jesque

thanks for merging this one can you give me a ping on slack as soon as you released it please?

**Pull request merged**